### PR TITLE
feat(credentials): document JMX credentials, keyring storage, header passthrough

### LINKED ISSUE #64: [Task] Add docs for matchExpression JMX credentials
https://github.com/cryostatio/cryostat/pull/1000
https://github.com/cryostatio/cryostat/pull/1019
https://github.com/cryostatio/cryostat-web/pull/481

### PR DESCRIPTION
**Pull request by andrewazores**
Fixes #64

### CODE REVIEW

File: guides/_subsections/configure-jmx-credentials-storage.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+## [Configure JMX Credentials Storage](#configure-jmx-credentials-storage)
+
+Target JVM applications may require Cryostat to pass an authentication challenge before being able to communicate over JMX
+and manage JFR. Cryostat has two supported mechanisms for these JMX credentials:
+<ol>
+  <li>
+    <b>JMX Credentials Keyring</b>: see <a href="#store-jmx-credentials">Store JMX Credentials</a> for more detail.
+    This mechanism entails uploading a JMX Credentials definition to the Cryostat backend keyring storage. Cryostat
+    automatically checks the keyring for credentials matching a target application when a request to that application
+    is opened. If no credentials are found, Cryostat responds to the requesting client (ex. the Cryostat Web UI) with
```
Comment (maxcao13) [inline guides/_subsections/configure-jmx-credentials-storage.md:10]: Not sure, but I would change ex. to e.g.

File: guides/_subsections/configure-jmx-credentials-storage.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    a response indicating the JMX authentication failure.
+  </li>
+  <li>
+    <b>X-JMX-Authorization</b> header passthrough: when the Cryostat Web UI receives a JMX authentication failure response
+    from the Cryostat server it displays a prompt asking for JMX credentials. Depending on the configuration set in this
```
Comment (maxcao13) [inline guides/_subsections/configure-jmx-credentials-storage.md:15]: `from the Cryostat server it displays` -> `from the Cryostat server, it displays`

File: guides/_subsections/configure-jmx-credentials-storage.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    section, those credentials may either be held locally in your browser session, or they may be uploaded to the Cryostat
+    credentials keyring. If the credentials are held locally in your browser session then the Cryostat Web UI will
```
Comment (maxcao13) [inline guides/_subsections/configure-jmx-credentials-storage.md:17]: `browser session then the` -> `browser session, then the`

File: guides/_subsections/configure-jmx-credentials-storage.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    automatically attach a custom header with API requests containing the JMX credentials you specify. The Cryostat
+    server sees this header and uses the credentials when opening the JMX connection to the target, but <i>does not
+    persist these credentials</i>. When the JMX credentials passthrough header is present they supersede credentials
```
Comment (maxcao13) [inline guides/_subsections/configure-jmx-credentials-storage.md:20]: `is present they supsersede` -> `is present, they supersede`

File: guides/_subsections/store-jmx-credentials.md [hunk 5]
```diff
@@ -1,42 +1,46 @@
...
 
-Cryostat stores JMX credentials according to each target’s unique JMX service URL, also known as a connection URL. Even if the underlying JVM instance changes, the target alias changes, or the target application restarts, Cryostat will apply the stored JMX credentials to the connection URL that the credentials are associated with.
-
-Here’s how to manage stored JMX credentials with the Cryostat web UI.
+If you intend to use Cryostat <a href="#create-an-automated-rule">Automated Rules</a> then you should store target application JMX credentials in Cryostat's JMX credentials keyring, which is outlined below. Even if you do not use Automated Rules you may find it more convenient to store credentials in the keyring, so that the Cryostat web UI does not need to prompt you for credentials when you manually access JFR information about target applications.
```
Comment (maxcao13) [inline guides/_subsections/store-jmx-credentials.md:4]: If you intend to use Automated Rules`,`

Comment (maxcao13) [inline guides/_subsections/store-jmx-credentials.md:4]: Even if you do not use Automated Rules`,` you may find it ....

I would also consider ending the sentence on `credentials in the keyring`. `This way, the Cryostat web UI does not need...`

File: guides/_subsections/store-jmx-credentials.md [hunk 6]
```diff
@@ -1,42 +1,46 @@
...
-      image-name="navigate-to-recordings-1.png"
+      image-name="navigate-to-recordings-1-220.png"
       text="
-        Alternatively, JMX credentials will also be stored if you navigate to either the <i>Recordings</i> tab or the <i>Events</i> tab and select a target JVM with JMX authentication enabled. The authentication form will appear, prompting you to enter your JMX credentials. The credentials will be automatically stored and will appear in the Stored JMX Credentials table.
+        Alternatively, JMX credentials may also be stored if you navigate to either the <i>Recordings</i> tab or the <i>Events</i> tab and select a target JVM with JMX authentication enabled. The authentication form will appear, prompting you to enter your JMX credentials. If you have <a href='#configure-jmx-credentials-storage'>configured Backend JMX Credentials Storage</a> then a new credentials definition with the provided username and password will be stored for this specific target application in the Cryostat keyring. If you have configured JMX credentials with the <i>Session</i> option then these entered credentials will <i>not</i> be stored in the Cryostat keyring.
```
Comment (maxcao13) [inline guides/_subsections/store-jmx-credentials.md:34]: If you have configured Backend JMX Credentials`,` then ...

**Comment by maxcao13** [comment]
I hope this is not actually being edited later and I did unnecessary work...

**Comment by andrewazores** [comment]
No, there would be no later editing stage for these upstream docs. We only have those resources for downstream product documentation and Red Hat blog posts.

**Review by maxcao13 — APPROVED** [review]
Looks good!

**Review by tthvo — APPROVED** [review]
Looks good! Also, looks like we have a new tab for Agent view coming up. Should we hold off until its merged and update the screenshots?

**Comment by andrewazores** [comment]
Most of the open PRs now are to merge into the upstream `version-220` branch, not straight into `main`. `version-220` will merge into `main` in a few weeks when we publish the upstream container images. I think I will start merging those PRs into `version-220` so we don't run into a bunch of conflicts and requiring work to update screenshots across different open PRs - the screenshot updates can all be done as a single PR against `version-220` later on, when that JMC Agent Plugin work is done and merged.

**Comment by tthvo** [comment]
Right thanks! That makes sense :D

**Pull request merged**